src/PowerShellEditorServices/Services/DebugAdapter/DebugService.cs

```diff
@@ -994,10 +994,45 @@ internal async void OnDebuggerStopAsync(object sender, DebuggerStopEventArgs e)
                     // Augment the top stack frame with details from the stop event
                     if (invocationTypeScriptPositionProperty.GetValue(e.InvocationInfo) is IScriptExtent scriptExtent)
                     {
-                        stackFrameDetails[0].StartLineNumber = scriptExtent.StartLineNumber;
-                        stackFrameDetails[0].EndLineNumber = scriptExtent.EndLineNumber;
-                        stackFrameDetails[0].StartColumnNumber = scriptExtent.StartColumnNumber;
-                        stackFrameDetails[0].EndColumnNumber = scriptExtent.EndColumnNumber;
+                        StackFrameDetails targetFrame = stackFrameDetails[0];
+
+                        // Certain context changes (like stepping into the default value expression
+                        // of a parameter) do not create a call stack frame. In order to represent
+                        // this context change we create a fake call stack frame.
+                        if (!string.IsNullOrEmpty(scriptExtent.File)
+                            && !PathUtils.IsPathEqual(scriptExtent.File, targetFrame.ScriptPath))
+                        {
+                            await debugInfoHandle.WaitAsync().ConfigureAwait(false);
```

Comment:
Normally I'd prefer to push this into the code for `FetchStackFramesAndVariablesAsync` but that would require giving it more context than it typically needs. Since this should be an edge case rather than the norm, I lean towards this being preferable.

That said I'm open to switching it up and just passing more context to fetch.